tasks/main.yml

```diff
@@ -1,21 +1,28 @@
 ---
-- name: centos, RHEL | Install EPEL
-  package:
-    name: epel-release
-  when: ansible_os_family == "RedHat" or ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
+- block:
+  - name: centos, RHEL | Install EPEL
+    package:
+      name: epel-release
+    when: ansible_os_family == "RedHat" or ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux'
 
-- name: Install common packages
-  package:
-    name: openvpn
-    update_cache: yes
+  - name: Install common packages
+    package:
+      name: openvpn
+      update_cache: yes
 
-- include_tasks: server.yml
-  when: inventory_hostname in openvpn_servers
+  - include_tasks: server.yml
+    when: inventory_hostname in openvpn_servers
 
-- include_tasks: client.yml
+  - include_tasks: client.yml
+    when:
+      - inventory_hostname in openvpn_clients
+      - inventory_hostname != openvpn_server
+    loop: "{{ openvpn_servers }}"
```

Comment:
Might be interesting to apply your `difference` filter trick to remove a predicate term but it is not significant.